test/unit/error.spec.ts

```diff
@@ -292,3 +285,88 @@ test("return from nested finally", () => {
     `;
     expect(util.transpileAndExecute(code)).toBe("finally AB");
 });
+
+test("throw and catch custom error object", () => {
+    util.testFunction`
+        try {
+            throw { x: "Hello error object!" };
```

Comment:
Would it make sense to add tests for other thrown types (`boolean`, `number` and `undefined`)?

Reply:
Thats actually a good idea. Especially `string` should be tested as I just noticed that Lua adds the source's location by default.